fpm/src/fpm_filesystem.f90

```diff
@@ -80,67 +80,98 @@ end function basename
 !!
 !! To be replaced by realpath/_fullname in stdlib_os
 !!
-function canon_path(path) result(canon)
-    character(*), intent(in) :: path
-    character(:), allocatable :: canon
+!! FIXME: Lot's of ugly hacks following here
+function canon_path(path)
```

Comment:
I'm not proud of what I've done here, but it seems to get the job done for now.

Reply:
I think adding in ugly hacks would make me less nervous if you also added some unit tests. 😉